updated SSO docs (#14279)
docker/docker.github.io: single-sign-on/index.md

When SSO is enabled, users are redirected to your provider’s authentication page to authenticate using SSO. They cannot authenticate using their personal login credentials (Docker ID and password). Docker currently supports Service Provider Initiated SSO flow. Your users must sign into Docker Hub or Docker Desktop to initiate the SSO authentication process.

Before enabling SSO in Docker Hub, administrators must work with their identity provider to configure their IdP to work with Docker Hub. Docker provides the Assertion Consumer Service (ACS) URL and the Entity ID. Administrators use this information to establish a connection between their IdP server and Docker Hub.

After establishing the connection between the IdP server and Docker Hub, administrators log into the organization in Docker Hub and complete the SSO enablement process. See the section **Enable SSO in Docker Hub** for detailed instructions.

> To enable SSO in Docker Hub, you need the following:

Reviewer: ```suggestion
To enable SSO in Docker Hub, you need the following information from your identity provider:
```

> * **SAML 2.0**: Entity ID, ACS URL, Single Logout URL and X.509 certificate

Reviewer: ```suggestion
* **SAML 2.0**: Single Sign-On URL and the X.509 signing certificate
```

    > **Note:**
    > the NameID is your email address and is set as the default.

>     > For example, <Subject><NameID>yourname@mycompany.com</NameID>. We also support the optional name attribute key; you must use lower-case text only.

Reviewer: ```suggestion
    > For example, <Subject><NameID>yourname@mycompany.com</NameID>. We also support the optional `name` attribute. This attribute name must be lower-cased. _The following is an example of this attribute in Okta._
```

Author: Thanks Josh!
